middleware/logger/logger.go

```diff
@@ -198,109 +234,13 @@ func New(config ...Config) fiber.Handler {
-			case TagLatency:
-				return buf.WriteString(fmt.Sprintf("%7v", stop.Sub(start).Round(time.Millisecond)))
-			case TagBody:
-				return buf.Write(c.Body())
-			case TagBytesReceived:
-				return appendInt(buf, len(c.Request().Body()))
-			case TagBytesSent:
-				return appendInt(buf, len(c.Response().Body()))
-			case TagRoute:
-				return buf.WriteString(c.Route().Path)
-			case TagStatus:
-				if cfg.enableColors {
-					return buf.WriteString(fmt.Sprintf("%s %3d %s", statusColor(c.Response().StatusCode(), colors), c.Response().StatusCode(), colors.Reset))
-				}
-				return appendInt(buf, c.Response().StatusCode())
-			case TagResBody:
-				return buf.Write(c.Response().Body())
-			case TagReqHeaders:
-				reqHeaders := make([]string, 0)
-				for k, v := range c.GetReqHeaders() {
-					reqHeaders = append(reqHeaders, k+"="+v)
-				}
-				return buf.Write([]byte(strings.Join(reqHeaders, "&")))
-			case TagQueryStringParams:
-				return buf.WriteString(c.Request().URI().QueryArgs().String())
-			case TagMethod:
-				if cfg.enableColors {
-					return buf.WriteString(fmt.Sprintf("%s %-7s %s", methodColor(c.Method(), colors), c.Method(), colors.Reset))
-				}
-				return buf.WriteString(c.Method())
-			case TagBlack:
-				return buf.WriteString(colors.Black)
-			case TagRed:
-				return buf.WriteString(colors.Red)
-			case TagGreen:
-				return buf.WriteString(colors.Green)
-			case TagYellow:
-				return buf.WriteString(colors.Yellow)
-			case TagBlue:
-				return buf.WriteString(colors.Blue)
-			case TagMagenta:
-				return buf.WriteString(colors.Magenta)
-			case TagCyan:
-				return buf.WriteString(colors.Cyan)
-			case TagWhite:
-				return buf.WriteString(colors.White)
-			case TagReset:
-				return buf.WriteString(colors.Reset)
-			case TagError:
-				if chainErr != nil {
-					return buf.WriteString(chainErr.Error())
-				}
-				return buf.WriteString("-")
-			default:
-				// Check if we have a value tag i.e.: "reqHeader:x-key"
-				switch {
-				case strings.HasPrefix(tag, TagReqHeader):
-					return buf.WriteString(c.Get(tag[10:]))
-				case strings.HasPrefix(tag, TagHeader):
-					return buf.WriteString(c.Get(tag[7:]))
-				case strings.HasPrefix(tag, TagRespHeader):
-					return buf.WriteString(c.GetRespHeader(tag[11:]))
-				case strings.HasPrefix(tag, TagQuery):
-					return buf.WriteString(c.Query(tag[6:]))
-				case strings.HasPrefix(tag, TagForm):
-					return buf.WriteString(c.FormValue(tag[5:]))
-				case strings.HasPrefix(tag, TagCookie):
-					return buf.WriteString(c.Cookies(tag[7:]))
-				case strings.HasPrefix(tag, TagLocals):
-					switch v := c.Locals(tag[7:]).(type) {
-					case []byte:
-						return buf.Write(v)
-					case string:
-						return buf.WriteString(v)
-					case nil:
-						return 0, nil
-					default:
-						return buf.WriteString(fmt.Sprintf("%v", v))
-					}
-				}
+			if logFunc, ok := tagMap[tag]; ok {
+				return logFunc(buf, c, w, tag)
+			}
+			split := strings.SplitAfter(tag, ":")
```

Comment:
this is a hot path where we should use other solutions for the check

split creates a new string, which then leads to new allocations and thus to performance degradation

we should use string.index to see where the ":" character is and then slice the string until then to use for the check in the tag map

Reply:
GET